libsol/parser.c

```diff
@@ -122,7 +122,22 @@ int parse_hash(Parser* parser, const Hash** hash) {
     return 0;
 }
 
+int parse_version(Parser* parser, MessageHeader* header) {
+    BAIL_IF(check_buffer_length(parser, 1));
+    const uint8_t version = *parser->buffer;
+    if (version & 0x80) {
+        header->versioned = true;
+        header->version = version & 0x7f;
+        advance(parser, 1);
+    } else {
+        header->versioned = false;
+        header->version = 0;
+    }
+    return 0;
+}
+
 int parse_message_header(Parser* parser, MessageHeader* header) {
+    BAIL_IF(parse_version(parser, header));
```

Comment:
we also need to conditionally parse the [ALU accounts `Vec`](https://github.com/solana-labs/solana/blob/master/sdk/program/src/message/versions/v0/mod.rs#L86), otherwise the "full message buffer consumed" check will fail.

additionally, we'll have to fail or force blind signing (if setting enabled) for any non-zero value. ALU account contents can't be verified by the app, so we don't really have a choice

Reply:
yes, this is how it works. ALU accounts are checked here:
https://github.com/LedgerHQ/app-solana/pull/48/files#diff-519d25b1ff94145ef8bf0cbcf7d71bb56f1a8b1472c9bf178c33c3332631619dR102
If the table is not empty then process_messsage_body() call fails, but it is still possible to continue if blind signing is enabled.
And if a transaction is unversioned or versioned with an empty ALU then transaction parsing works as before including "full message buffer consumed" check.